api/ruxos_posix_api/src/imp/fs.rs

```diff
@@ -686,10 +686,39 @@ pub fn sys_faccessat(dirfd: c_int, pathname: *const c_char, mode: c_int, flags:
 
 /// changes the current working directory to the directory specified in path.
 pub fn sys_chdir(path: *const c_char) -> c_int {
-    let p = char_ptr_to_str(path).unwrap();
+    let p = char_ptr_to_absolute_path(path).unwrap();
     debug!("sys_chdir <= path: {}", p);
     syscall_body!(sys_chdir, {
         set_current_dir(p)?;
         Ok(0)
     })
 }
+
+/// from char_ptr get absolute_path_str
+pub fn char_ptr_to_absolute_path<'a>(ptr: *const c_char) -> LinuxResult<&'a str> {
+    if ptr.is_null() {
+        Err(LinuxError::EFAULT)
+    } else {
+        unsafe {
+            let cstr = CStr::from_ptr(ptr);
+            let path: &str = cstr.to_str().map_err(|_| LinuxError::EINVAL)?;
+            if let Some(stripped) = path.strip_prefix("..") {
+                let mut cwd = ruxfs::api::current_dir()?;
+                if let Some(index) = cwd.rfind('/') {
+                    cwd.truncate(index);
+                    if let Some(index) = cwd.rfind('/') {
+                        cwd.truncate(index);
+                    }
+                }
+                let absolute_path: String = cwd + stripped;
+                return Ok(Box::leak(absolute_path.into_boxed_str()));
+            } else if let Some(stripped) = path.strip_prefix('.') {
+                let cwd = ruxfs::api::current_dir()?;
+                let absolute_path: String = cwd + stripped;
+                return Ok(Box::leak(absolute_path.into_boxed_str()));
```

Comment:
**It will cause memory leaky there**, try to use change the type of return value instead.

Reply:
I use `borrow::Cow` to fix this promblem.